src/components/mc/cuda/mc_cuda.h

```diff
@@ -136,4 +136,21 @@ extern ucc_mc_cuda_t ucc_mc_cuda;
 #define MC_CUDA_CONFIG                                                         \
     (ucc_derived_of(ucc_mc_cuda.super.config, ucc_mc_cuda_config_t))
 
+#define UCC_MC_CUDA_INIT_STREAM() do {                                         \
+    if (ucc_mc_cuda.stream == NULL) {                                          \
+        cudaError_t cuda_st;                                                   \
+        ucc_spin_lock(&ucc_mc_cuda.init_spinlock);                             \
+        if (ucc_mc_cuda.stream == NULL) {                                      \
+            cuda_st = cudaStreamCreateWithFlags(&ucc_mc_cuda.stream,           \
```

Comment:
should we create high_priority stream with cudaStreamCreateWithPriority()

Reply:
yes, but I believe it should be an option rather than creating high priority steam always.
Do you mean make it part of this PR?

Reply:
lets do it in another PR